### PR TITLE
Reduce colreaders

### PR DESCRIPTION
**Pull request by ccfelius**
The ReadStatistics call was creating a root_reader for every column, now we do this only once at initialization.
This reduces the number of creations by the amount of columns (PropagateStatistics does this for all columns)

### CODE REVIEW

File: extension/parquet/parquet_reader.cpp [hunk 1]
```diff
@@ -399,7 +399,6 @@ void ParquetReader::InitializeSchema() {
 	if (file_meta_data->schema.size() < 2) {
 		throw FormatException("Need at least one non-root column in the file");
 	}
-	auto root_reader = CreateReader();
```
Comment (Mytherin) [inline extension/parquet/parquet_reader.cpp:402]: Can this not stay in the `InitializeSchema`? It seems like we are initializing the `root_reader` only right before calling `InitializeSchema` anyway.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me. One minor comment:

File: extension/parquet/parquet_reader.cpp (hunk 1 above)
Comment (ccfelius) [reply extension/parquet/parquet_reader.cpp:402]: That's right, moved it to `InitializeSchema()`. Thanks!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**